src/utilities.ts

```diff
@@ -165,8 +169,9 @@ export async function getRegistryContract(
 ) {
     if (!isValidVersion(version)) throw UNSUPPORTED_VERSION_ERROR
 
+    const router = await getRouterContract(version, wallet)
```

Comment:
Not a requested change but just fyi you could have done `(await getRouterContract(version, wallet)).getRegistryAddress()`